tests/perftest.py

```diff
@@ -16,6 +16,51 @@
 import utils
 from phonetest import PhoneTest, PhoneTestResult
 
+"""
+   PerfherderArtifact and PerfherderSuite and specific formats for
+   Perfherder as defined in:
+   https://bugzilla.mozilla.org/show_bug.cgi?id=1175295
+
+   These work with summarized suite values and summarized test value,
+   instead of working with raw replicates.
+
+   In the future it would be nice to have a generic PerfData class which
+   stores the raw replicates and then the output functions can do the final
+   summarization and calculations as defined by the output medium (e.g. perfherder)
+"""
+class PerfherderArtifact():
+    def __init__(self, suites=[]):
+        self._framework = 'autophone'
+        self._suites = suites
+
+    def addSuite(self, suite):
+        self._suites.append(suite)
+
+    def toJSON(self):
+        return {'framework': self._framework,
+               'suites': [s.toJSON() for s in self._suites]}
+
+
+class PerfherderSuite():
+    def __init__(self, name=None, value=0, subtests=[]):
```

Comment:
You should provide an array as a default parameter to a namaed argument, as they are mutable. Use the pattern described here instead:

http://stackoverflow.com/a/6435828